sdk/table/azure-table/azure/table/_table_client.py

```diff
@@ -0,0 +1,575 @@
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        if_match, if_not_match = _get_match_headers(kwargs=dict(kwargs, etag=etag, match_condition=match_condition),
+                                                    etag_param='etag', match_param='match_condition')
+
+        if table_entity_properties:
+            partition_key = table_entity_properties['PartitionKey']
+            row_key = table_entity_properties['RowKey']
+            table_entity_properties = _add_entity_properties(table_entity_properties)
+
+        try:
+            self._client.table.update_entity(
+                table=self.table_name,
+                partition_key=partition_key,
+                row_key=row_key,
+                table_entity_properties=table_entity_properties,
+                if_match=if_match or if_not_match or "*",
+                **kwargs)
+        except ResourceNotFoundError:
+            raise ResourceNotFoundError
+
+    @distributed_trace
+    def merge_entity(
+            self,
+            partition_key=None,  # type: str
+            row_key=None,  # type: str
+            etag=None,  # type: str
+            match_condition=None,  # type: MatchConditions
+            table_entity_properties=None,  # type: Optional[Dict[str, object]]
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        """Merge entity in a table.
+
+        :param partition_key: The partition key of the entity.
+        :type partition_key: str
+        :param row_key: The row key of the entity.
+        :type row_key: str
+         :param match_condition: MatchCondition
+        :type match_condition: ~azure.core.MatchConditions
+        :param etag: Etag of the entity
+        :type etag: str
+        :param table_entity_properties: The properties for the table entity.
+        :type table_entity_properties: dict[str, object]
+        :keyword callable cls: A custom type or function that will be passed the direct response
+        :return: None, or the result of cls(response)
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        if_match, if_not_match = _get_match_headers(kwargs=dict(kwargs, etag=etag, match_condition=match_condition),
+                                                    etag_param='etag', match_param='match_condition')
+
+        if table_entity_properties:
+            partition_key = table_entity_properties['PartitionKey']
+            row_key = table_entity_properties['RowKey']
+            table_entity_properties = _add_entity_properties(table_entity_properties)
+
+        try:
+            self._client.table.merge_entity(table=self.table_name, partition_key=partition_key,
+                                            row_key=row_key, if_match=if_match or if_not_match or '*',
+                                            table_entity_properties=table_entity_properties, **kwargs)
+
+        except ResourceNotFoundError:
```

Comment:
Let's check if we need this, the generated code might throw the appropriate error already.